core/src/main/clojure/xtdb/operator/scan.clj

```diff
@@ -82,34 +82,34 @@
                          (time/instant->micros))))]
 
       (when-let [system-time (some-> at-tx (.getSystemTime) time/instant->micros)]
-        (.lte (.getSystemFrom bounds) system-time)
+        (.lte (.getSystemTime bounds) system-time)
 
         (when-not for-system-time
-          (.gt (.getSystemTo bounds) system-time)))
+          (.gte (.getSystemTime bounds) system-time)))
 
-      (letfn [(apply-constraint [constraint ^TemporalBounds$TemporalColumn start-col, ^TemporalBounds$TemporalColumn end-col]
+      (letfn [(apply-constraint [constraint ^TemporalDimension temporal-dimension]
                 (when-let [[tag & args] constraint]
                   (case tag
                     :at (let [[at] args
                               at-μs (->time-μs at)]
-                          (.lte start-col at-μs)
-                          (.gt end-col at-μs))
+                          (.lte temporal-dimension at-μs)
+                          (.gte temporal-dimension at-μs))
 
                     ;; overlaps [time-from time-to]
                     :in (let [[from to] args]
```

Comment:
Would be good to clarify what overlap exactly means here. I understand it as the intervals need to have a `valid-from` strictly less than the `from` from `:in`.